Sources/StreamChat/ChatClient.swift

```diff
@@ -404,6 +405,48 @@ public class ChatClient {
             completion()
         }
     }
+    
+    // MARK: - Listening for Client Events
+    
+    /// Subscribes to web-socket events of the specified event type.
+    ///
+    /// - Note: The handler is always called on the main thread.
+    ///
+    /// An example of observing connection status changes:
+    /// ```swift
+    /// client.subscribe(to: ConnectionStatusUpdated.self) { connectionEvent in
+    ///     switch connectionEvent.connectionStatus {
+    ///         case .connected:
+    ///           …
+    ///     }
+    /// }
+    /// ```
+    ///
+    /// - SeeAlso: ``Chat.subscribe(to:handler:)`` for subscribing to channel specific events.
+    ///
+    /// - Parameters:
+    ///   - event: The event type to subscribe to (e.g. ``ConnectionStatusUpdated``).
+    ///   - handler: The handler closure which is called when the event happens.
+    ///
+    /// - Returns: A cancellable instance, which you use when you end the subscription. Deallocation of the result will tear down the subscription stream.
+    @available(iOS 13.0, *)
+    public func subscribe<T>(to eventType: T.Type, handler: @escaping (T) -> Void) -> AnyCancellable where T: Event {
+        eventNotificationCenter.subscribe(to: T.self, handler: handler)
+    }
+
+    /// Subscribes to all the web-socket events.
+    ///
+    /// - SeeAlso: ``Chat.subscribe(handler:)`` for subscribing to channel specific events.
+    ///
+    /// - Parameter handler: The handler closure which is called when the event happens.
+    ///
+    /// - Returns: A cancellable instance, which you use when you end the subscription. Deallocation of the result will tear down the subscription stream.
+    @available(iOS 13.0, *)
+    public func subscribe(_ handler: @escaping (Event) -> Void) -> AnyCancellable {
```

Comment:
I don't think we need this either 🤔 If a customer creates a custom event type, they will also be able to observe it with the other one right? If that is true (I think it is), then I would rather not have this one. It just causes more confusion to the customers, having 2 ways to do the same thing

Reply:
You are right, this one receives every single channel event as well.

Reply:
client.subscribe { event in
  switch event {
    case  let connectionEvent as ConnectionStatusUpdated:
      ...
  }
}

client.subscribe { event in
  Logger().debug(type(of: event))
}